flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaSerializerWrapper.java

```diff
@@ -33,25 +33,37 @@
 import static org.apache.flink.util.Preconditions.checkState;
 
 class KafkaSerializerWrapper<IN> implements SerializationSchema<IN> {
+
     private final Class<? extends Serializer<? super IN>> serializerClass;
+    // whether the serializer is for key or value
+    private final boolean isKey;
     private final Map<String, String> config;
     private final Function<? super IN, String> topicSelector;
 
     private transient Serializer<? super IN> serializer;
 
     KafkaSerializerWrapper(
             Class<? extends Serializer<? super IN>> serializerClass,
+            boolean isKey,
             Map<String, String> config,
             Function<? super IN, String> topicSelector) {
         this.serializerClass = checkNotNull(serializerClass);
+        this.isKey = isKey;
         this.config = checkNotNull(config);
         this.topicSelector = checkNotNull(topicSelector);
     }
 
+    KafkaSerializerWrapper(
+            Class<? extends Serializer<? super IN>> serializerClass,
+            Map<String, String> config,
+            Function<? super IN, String> topicSelector) {
+        this(serializerClass, false, config, topicSelector);
+    }
+
     KafkaSerializerWrapper(
             Class<? extends Serializer<? super IN>> serializerClass,
             Function<? super IN, String> topicSelector) {
-        this(serializerClass, Collections.emptyMap(), topicSelector);
+        this(serializerClass, false, Collections.emptyMap(), topicSelector);
```

Comment:
This constructor is invoked by `KafkaRecordSerializationSchemaBuilder::setKafkaKeySerializer(...)`. It is probably not intuitive to set `isKey=false` here.